src/Router/AppShell.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { SFC } from "react"
+import serialize from "serialize-javascript"
+import { ContextProvider } from "Components/Artsy"
+
+interface Props {
+  loadableState?: any
+  relayData?: any
+  provide?: any
+}
+
+export const AppShell: SFC<Props> = props => {
+  const { loadableState, relayData, children, provide } = props
+
+  return (
+    <ContextProvider {...provide}>
+      <div>
+        <div
+          // FIXME: type
+          {...{ suppressHydrationWarning: true }}
+          dangerouslySetInnerHTML={{
+            __html: `
+            ${loadableState ? loadableState.getScriptTag() : ""}
+
+            <script>
+              var __RELAY_BOOTSTRAP__ = ${serialize(
+                JSON.stringify(relayData || {}),
+                {
+                  isJSON: true,
+                }
+              )};
+            </script>
```

Comment:
I came up with this approach when working on stitch's [`componentRenderer`](https://github.com/artsy/stitch/blob/master/src/componentRenderer.js#L35) and realized that we can just inline SSR bootstrap data directly in containing div rather than having to worry about the modifying the overall page shell. It would throw warnings due to clientside rehydration mismatch, but I learned tonight that React has a light escape hatch via `suppressHydrationWarning`; I just need to figure out how to type the div correctly so that TS doesn't get angry.

Reply:
Clever. I had some initial reservations about this, but looking [in the docs](https://reactjs.org/docs/react-dom.html#hydrate), it seems like it's fine. It only ignores the warning at that level so any hydration warnings for children _should_ be preserved. 

I think longer term hoisting it out of the AppShell would be ideal though.

Reply:
for sure, I think once we replace the outer page shell with react and establish a proper line of communication thats def the way to go